### DISCUSSION TITLE
Reverse proxy: do Client TLS before Server TLS

### DISCUSSION THREAD

**Discussion by oliverrahner** [mitmproxy/mitmproxy#6282] [Q&A] [answered]
Hi,

I want to select the client certificate for the server connection based on the client certificate of the client connection.
I did already manage to retrieve information about the client certificate of the client connection, but because the default layer stack for reverse proxy mode does server TLS first, I don't have this information available at the time of server TLS initialization...

I tried fiddling around with the `next_layer` hook, but it is only called after the `ServerTLSLayer` has already been processed (`tls_start_server` is called before `next_layer`).

So is what I want to do even feasible or are there dependencies I didn't see that would prevent this from working?
Any hints on how to do what I'm looking for?

**Comment by mhils** [comment, marked as answer]
Have you tried setting connection_strategy to lazy? :)

**Reply by oliverrahner** [reply]
🤦‍♂️
I had looked through the available options, but I must have missed that one... Will try if that suits my purpose :)

**Reply by oliverrahner** [reply]
That was way too easy :)
Thank you very much!
